contractcourt/anchor_resolver.go

```diff
@@ -0,0 +1,167 @@
+package contractcourt
+
+import (
+	"errors"
+	"io"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/sweep"
+)
+
+// anchorResolver is a resolver that will attempt to sweep our anchor output.
+type anchorResolver struct {
+	// anchorSignDescriptor contains the information that is required to
+	// sweep the anchor.
+	anchorSignDescriptor input.SignDescriptor
+
+	// anchor is the outpoint on the commitment transaction.
+	anchor wire.OutPoint
+
+	// resolved reflects if the contract has been fully resolved or not.
+	resolved bool
+
+	// broadcastHeight is the height that the original contract was
+	// broadcast to the main-chain at. We'll use this value to bound any
+	// historical queries to the chain for spends/confirmations.
+	broadcastHeight uint32
+
+	// chanPoint is the channel point of the original contract.
+	chanPoint wire.OutPoint
+
+	contractResolverKit
+}
+
+// newAnchorResolver instantiates a new anchor resolver.
+func newAnchorResolver(anchorSignDescriptor input.SignDescriptor,
+	anchor wire.OutPoint, broadcastHeight uint32,
+	chanPoint wire.OutPoint, resCfg ResolverConfig) *anchorResolver {
+
+	r := &anchorResolver{
+		contractResolverKit:  *newContractResolverKit(resCfg),
+		anchorSignDescriptor: anchorSignDescriptor,
+		anchor:               anchor,
+		broadcastHeight:      broadcastHeight,
+		chanPoint:            chanPoint,
+	}
+
+	r.initLogger(r)
+
+	return r
+}
+
+// ResolverKey returns an identifier which should be globally unique for this
+// particular resolver within the chain the original contract resides within.
+func (c *anchorResolver) ResolverKey() []byte {
+	// The anchor resolver is stateless and doesn't need a database key.
+
+	return nil
+}
+
+// Resolve offers the anchor output to the sweeper and waits for it to be swept.
+//
+// NOTE: This function MUST be run as a goroutine.
+func (c *anchorResolver) Resolve() (ContractResolver, error) {
+	// Attempt to update the sweep parameters to the post-confirmation
```

Comment:
shuold have waitgroup?

Reply:
There is a waitgroup for this goroutine, the add happens in `ChannelArbitrator.launchResolvers`

Reply:
Ok looking at the callsite I think the comment `NOTE: This function MUST be run as a goroutine.` is wrong, so I wouldn't mind removing it. But this is also pre-existing.

Reply:
Yes indeed. It is up to the caller to decide how to run this. And as you say, the call site already says it should be run as a goroutine

Reply:
Removed comment